### PR TITLE
LPS-197771 Create generic creation button actions

### PR DESCRIPTION
**Pull request by markocikos**
## Info
Related Story/Task: [LPS-186875](https://liferay.atlassian.net/browse/LPS-186875) / [LPS-197771](https://liferay.atlassian.net/browse/LPS-197771)
Related documentation: [Research for actions confluence page](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2418148860/LPS-186947+-+Tech+research+for+actions+in+dataset+views)
UI: [Creation Actions](https://www.figma.com/file/ThsIousIfUrvJXgy4bfX3j/LPS-193209-design-how-to-define-generic-actions?type=design&node-id=201-55288&mode=design&t=lSok6SosJ9wQFJdL-0)
Feature Flag: `LPS-194395`
Followup on https://github.com/liferay-frontend/liferay-portal/pull/3727

## Story
We need to allow admin users to define actions that apply to the items in a Dataset (actions in the :heavy_plus_sign: button in the FDS component).
The story covers the addition of new item actions (modal & side panel) as well as edition, deletion and sorting of both types of actions.

## UI

[Screencast from 2023-10-05 10-25-38.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/b740617d-f153-4f8a-b220-00bcef7b16e8)

## Concerns & Questions
- Naming. Find some differences between the research document and the design implementation
- ~~`ItemActionForm` could be reused so we could rename to a generic `ActionForm` (not renamed to allow clear review)~~ **done**
- Mixing of tabs (number index based) and sections (text based) make UI complex. I need to think of a better way to handle it.
- Need to change `<ClaySelectWithOption>` by `<Picker>` to match the design
- Do we need to validate `url` format?

### CODE REVIEW

**Comment by markocikos** [comment]
@juanjofgliferay I created a followup PR because I couldn't push to your branch. Please invite me as a collaborator on your fork, I invited you on mine.

I made two changes, in two additional commits:
1. Trigger of validation on change, instead of on blur. See [relevant discussion on Slack](https://liferay.slack.com/archives/C3JBR21HA/p1697037847907879).
2. I noticed that tab switching does not work correctly in the latest version. This was caused by [the change not to use useEffect](https://github.com/liferay-frontend/liferay-portal/pull/3727#discussion_r1352796158). After the change, we did validation on the old state.

**Comment by markocikos** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ const Actions = ({fdsView, namespace, spritemap}: IFDSViewSectionProps) => {
 		return breadcrumbItems;
 	};
 
-	const loadFDSActions = async () => {
+	const loadFDSActions = async ({activeTab}: {activeTab: number}) => {
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx:118]: Good catch. I didn't realize that it was not working as expected

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Comment by markocikos** [comment]
@juanjofgliferay @JerryAdrianoo I pushed update to the test that failed in https://github.com/liferay-frontend/liferay-portal/pull/3749#issuecomment-1759984581, as well as lang key cleanup. The test failed because we changed label in this PR.

**Comment by markocikos** [comment]
ci:forward

**Comment by JerryAdrianoo** [comment]
@markocikos, as you said, these failures are not related.
[Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1697219928420&testrayRunIdA=2440865274&testrayRunIdB=2440978171)

**Comment by juanjofgliferay** [comment]
ci:forward:force

**Pull request closed without merging**